lldb/tools/lldb-dap/DAP.cpp

```diff
@@ -1209,13 +1213,89 @@ protocol::Capabilities DAP::GetCapabilities() {
 }
 
 void DAP::StartEventThread() {
-  event_thread = std::thread(&DAP::EventThread, this);
+  // Get event thread for this debugger (creates it if it doesn't exist)
+  event_thread_sp = DAPSessionManager::GetInstance().GetEventThreadForDebugger(
+      debugger, this);
 }
 
 void DAP::StartProgressEventThread() {
   progress_event_thread = std::thread(&DAP::ProgressEventThread, this);
 }
 
+llvm::Error DAP::StartEventThreads() {
+  if (clientFeatures.contains(eClientFeatureProgressReporting))
+    StartProgressEventThread();
+
+  StartEventThread();
+
+  return llvm::Error::success();
+}
+
+llvm::Error DAP::InitializeDebugger(std::optional<uint32_t> target_idx) {
+  // Initialize debugger instance (shared or individual)
+  if (target_idx) {
+    auto shared_debugger =
+        DAPSessionManager::GetInstance().GetSharedDebugger(*target_idx);
+    if (!shared_debugger) {
```

Comment:
I think this will never happen, or do you know of a specific case in which the debugger will get destroyed and the target kept alive?